packages/block-editor/src/utils/block-removal.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import {
+	Modal,
+	Button,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { __, _n } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../store';
+
+export const blockTypePromptMessages = {
+	'core/query': __( 'Query Loop displays a list of posts.' ),
+	'core/post-content': __(
+		'Post Content displays the content of a post or page.'
+	),
+};
+
+export function BlockRemovalWarningModal() {
+	const { removalFunction, blocksToPromptFor: blockNames } = useSelect(
+		( select ) => select( blockEditorStore ).isRemovalPromptDisplayed()
+	);
+
+	const { displayRemovalPrompt, removalPromptExists } =
+		useDispatch( blockEditorStore );
+
+	// Signalling the removal prompt is in place.
+	useEffect( () => {
+		removalPromptExists( true );
+		return () => {
+			removalPromptExists( false );
+		};
+	}, [ removalPromptExists ] );
+
+	if ( ! blockNames ) {
+		return;
+	}
+
+	const closeModal = () => displayRemovalPrompt( false );
+
+	const onConfirmRemoval = () => {
+		removalFunction();
```

Comment:
Apologies if I'm missing details here, but why is `removalFunction` required? Could we call `removeBlocks` (possibly with the force flag) with the selected clientIds instead of storing a `removalFunction` in state?

Reply:
Yeah, this is something I raised a couple of weeks ago, and we could just store the client IDs and the `selectPrevious` boolean.